var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -119,6 +145,10 @@ class Mpich(AutotoolsPackage):
     conflicts('pmi=pmi2', when='device=ch3 netmod=ofi')
     conflicts('pmi=pmix', when='device=ch3')
 
+    # Avoid identical installations with different specs.
```

Comment:
After https://github.com/spack/spack/pull/15235#issuecomment-598604206 I think I understand this comment: you are saying that `@:3.2~hydra+libxml2` and `@:3.2~hydra+lixml2` would result in the same `mpich` installation. I think it would be clearer to say:

> mpich does not require libxml2 for versions before 3.3 when ~hydra is set: prevent users from setting ~libxml2 in this case to avoid generating an identical mpich installation

If you agree, could you update this comment? If not, let me know and I think this is set to go.